obspy/io/hypodd/pha.py

```diff
@@ -111,19 +128,128 @@ def _read_pha(filename, inventory=None, id_map=None, id_default='.{}..{}',
     with io.open(filename, 'r', encoding=encoding) as f:
         text = f.read()
-    events = [_block2event(block, seed_map, id_default, ph2comp)
+    events = [_block2event(block, seed_map, id_default, ph2comp, eventid_map)
               for block in text.split('#')[1:]]
     return Catalog(events)
 
 
+PHA1 = ('# {o.time.year} {o.time.month:>2} {o.time.day:>2} {o.time.hour:>2}'
+        ' {o.time.minute:>2} {o.time.second:>2}.{o.time.microsecond:06d}  '
+        '{o.latitude} {o.longitude} {depth}  {mag}  {he} {ve} {rms}'
+        '   {evid:>9}\n')
+PHA2 = '{p.waveform_id.station_code:6}  {relt:.4f}  {weight}  {p.phase_hint}\n'
+
+
+def _map_eventid(evid, eventid_map, used_ids, counter):
+    idpha = evid
+    if evid in eventid_map:
+        idpha = eventid_map[evid]
+        if not idpha.isdigit() or len(idpha) > 9:
+            msg = ('Invalid value in eventid_map, pha event id has to be '
+                   'digit with max 9 digits')
+            raise ValueError(msg)
+        return idpha
+    if not idpha.isdigit():
+        idpha = ''.join(char for char in idpha if char.isdigit())
+    if len(idpha) > 9:
+        idpha = idpha[:9]
+    while idpha == '' or idpha in used_ids:
+        idpha = str(counter[0])
+        counter[0] += 1
+    if idpha != evid:
+        eventid_map[evid] = idpha
+    used_ids.add(idpha)
+    return idpha, eventid_map
+
+
+def _write_pha(catalog, filename, eventid_map=None,
+               **kwargs):  # @UnusedVariable
+    """
+    Write a HypoDD PHA file.
+
+    .. warning::
+        This function should NOT be called directly, it registers via the
+        the :meth:`~obspy.core.event.Catalog.write` method of an
+        ObsPy :class:`~obspy.core.event.Catalog` object, call this instead.
+
+    :type catalog: :class:`~obspy.core.event.catalog.Catalog`
+    :param catalog: The ObsPy Catalog object to write.
+    :type filename: str or file
+    :param filename: Filename to write or open file-like object.
+    :param dict eventid_map: Desired mapping of event resource ids (dict keys)
+        to hypodd event ids (dict values).
+        HYPODD expects integer event ids with maximal 9 digits. If the event
+        resource id is not present in the mapping,
+        the event resource id is stripped of all non-digit characters and
+        truncated to a length of 9 chars. If this method does not generate a
+        valid hypodd event id, a counter starting at 1000 is used.
+
+    :returns: Dictionary eventid_map with mapping of event resource id to
+        hypodd event id. Items are only present if both ids are different.
+    """
+    lines = []
+    if eventid_map is None:
+        eventid_map = {}
+    args_map_eventid = (eventid_map, set(eventid_map.values()), [1000])
+    for event in catalog:
+        try:
+            ori = event.preferred_origin() or event.origins[0]
+        except IndexError:
+            warn('Skip writing events with missing origin')
```

Comment:
```suggestion
            warn(f'Skipping writing event with missing origin: {event}')
```
I would probably want to know which events were being skipped.